apps/web/components/eventtype/EventAvailabilityTab.tsx

```diff
@@ -278,25 +279,168 @@ const EventTypeSchedule = ({ eventType }: { eventType: EventTypeSetup }) => {
   );
 };
 
+const TeamMemberSchedule = ({ host }: { host: Host }) => {
+  const { t } = useLocale();
+
+  const formMethods = useFormContext<FormValues>();
+  const { watch, setValue, getValues } = formMethods;
+  const hosts = getValues("hosts");
+
+  const getIdx = () =>
+    Math.max(
+      hosts.findIndex((_host) => _host.userId === host.userId),
+      0
+    );
+
+  const [options, setOptions] = useState<AvailabilityOption[]>([]);
+  const { data, isPending } = trpc.viewer.availability.schedule.getAllSchedulesByUserId.useQuery({
+    userId: host.userId,
+  });
+
+  useEffect(() => {
+    if (!data || isPending) {
+      return;
+    }
+
+    const schedules = data.schedules;
+    const scheduleOptions = schedules.map((schedule) => ({
+      value: schedule.id,
+      label: schedule.name,
+      isDefault: schedule.isDefault,
+      isManaged: false,
+    }));
+
+    setOptions(scheduleOptions);
+
+    const idx = getIdx();
+    //Set to defaultSchedule if Host Schedule is not previously selected
+    const scheduleId = getValues(`hosts.${idx}.scheduleId`);
+    const availability = getValues(`hosts.${idx}.availability`);
+    const value = scheduleOptions.find((option) =>
+      scheduleId
+        ? option.value === scheduleId
+        : availability
+        ? option.value === availability?.value
+        : option.value === schedules.find((schedule) => schedule.isDefault)?.id
+    );
+
+    if (!scheduleId) {
+      setValue(`hosts.${idx}.scheduleId`, value?.value || null, { shouldDirty: true });
+      setValue(`hosts.${idx}.availability`, value || null, { shouldDirty: true });
+    }
+
+    if (!availability) {
+      setValue(`hosts.${idx}.availability`, value || null, { shouldDirty: false });
+    }
+  }, [data]);
+
+  const idx = getIdx();
+  const availabilityValue = watch(`hosts.${idx}.availability`);
```

Comment:
No need to store the availabilityValue, you can find the selected option using scheduleId and schedules.

Reply:
used to set initial value, watch is not required. Updated.